stitch/src/androidTest/java/com/mongodb/stitch/StitchClientServiceTest.kt

```diff
@@ -186,4 +197,52 @@ class StitchClientServiceTest {
             assertThat(it.isSuccessful, it.exception)
         })
     }
+
+    /**
+     * Generates a HMAC for the given [data] with the given [key] using HMAC-SHA256.
+     *
+     * @returns HMAC
+     */
+    private fun createHmac(data: ByteArray, key: ByteArray): ByteArray {
+        val keySpec = SecretKeySpec(key, "HmacSHA256")
+        val mac = Mac.getInstance("HmacSHA256")
+        mac.init(keySpec)
+
+        val hmac = mac.doFinal(data)
+        return hmac
+    }
+
+    @Test
+    fun testCustomAuth() {
```

Comment:
[minor] We should use the library included to do this JWT instead of doing it ourselves.